### PR TITLE
[website] Refactor on existing Tutorials part structure

### PR DESCRIPTION
**Pull request by StrayBird-ATSH**
## Description ##
According to issue (#1081), the main structure of the website examples part needs to be refactored. Currently this pull request deals with part of this issue. _This pull request refactors all the existing files in the examples part of the website._ All existing structure of the examples part of the website are modified accordingly. For new features about adding new files to the examples part of the website, they will be handled in future pull requests.

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

cc @dmlc/gluon-nlp-team

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1154/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1154/6/index.html

**Comment by mli** [comment]
Job PR-1154/7 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1154/7/index.html

**Comment by StrayBird-ATSH** [comment]
After a pilot experiment and research, I have found that I could only add up to 3 cards in a cards container in the rst file.
Would you mind changing the structure a bit to limit to 3 cards or less; Or do you want me to do some more research to find out how to add more cards?
Please refer to the difference before and after commit [62f7eb0](https://github.com/dmlc/gluon-nlp/pull/1154/commits/62f7eb01892f749e1e6e691c118165f0094e08c3) to find out how it fails on four cards.
@eric-haibin-lin 

Thanks for your reply!

**Comment by eric-haibin-lin** [comment]
@StrayBird-ATSH what did you mean by it failed? Are you referring to this error msg? 
```
[2020-02-13T11:55:37.743Z] 

[2020-02-13T11:55:37.743Z] Exception occurred:

[2020-02-13T11:55:37.743Z]   File "/var/lib/jenkins/workspace/gluon-nlp-cpu-py3-master/conda/cpu/py3-master/lib/python3.5/site-packages/mxtheme/card.py", line 25, in run

[2020-02-13T11:55:37.743Z]     cid = nodes.make_id("card-{}".format(options['title']))

[2020-02-13T11:55:37.743Z] KeyError: 'title'

[2020-02-13T11:55:37.743Z] The full traceback has been saved in /tmp/sphinx-err-obhz33o5.log, if you want to report the issue to the developers.

[2020-02-13T11:55:37.743Z] Please also report this if it was a user error, so that a better error message can be provided next time.

[2020-02-13T11:55:37.743Z] A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!

[2020-02-13T11:55:38.303Z] Makefile:57: recipe for target 'html' failed

[2020-02-13T11:55:38.303Z] make[1]: *** [html] Error 2

[2020-02-13T11:55:38.303Z] make[1]: Leaving directory '/var/lib/jenkins/workspace/gluon-nlp-cpu-py3-master/docs'

[2020-02-13T11:55:38.303Z] Makefile:61: recipe for target 'docs' failed

[2020-02-13T11:55:38.303Z] make: *** [docs] Error 2
```

**Comment by eric-haibin-lin** [comment]
http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1154/7/examples/word_embedding/index.html
Looks like there's some rendering issue
Screen Shot 2020-02-13 at 9 21 35 PM

**Comment by StrayBird-ATSH** [comment]
Yes. I was referring to that error messages. That error prevents the entire website from being built. Even though it didn't mention what on earth caused the error, I gradually found out that it was the 4 cards causing it fail. It will compile successfully on 3 cards.

**Comment by StrayBird-ATSH** [comment]
> http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1154/7/examples/word_embedding/index.html
> Looks like there's some rendering issue
> Screen Shot 2020-02-13 at 9 21 35 PM

Thanks for pointing out this. This seems to be caused by a minor missing space problem. I have already fixed it in the latest commit. For now you can preview the website of the [6th commit in this pull request](http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1154/6/examples/word_embedding/index.html).

Thanks!

**Comment by eric-haibin-lin** [comment]
@szha are u aware of any limitation on the number of cards for the theme??

**Comment by StrayBird-ATSH** [comment]
> @szha are u aware of any limitation on the number of cards for the theme??

Sorry... It seems that it is the missing indentation among one of the four cards that prevents the cards from compilation. Now I finds it... The cause was so trivial... Now hopefully this problem can be solved.

Perhaps if I didn't put the original three cards on and let the indentation-caused rendering problem come out, I would never realize it was the indentation causing this problem.

**Comment by mli** [comment]
Job PR-1154/8 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1154/8/index.html

**Comment by mli** [comment]
Job PR-1154/9 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1154/9/index.html

**Comment by StrayBird-ATSH** [comment]
I think now this pull request should be ready for reviewing and merging.
Please give any advice if my refactor doesn't meet what you wish.

File: docs/examples/index.rst [hunk 1]
```diff
@@ -23,6 +23,20 @@ Word Embedding
       Learn how to train fastText and word2vec embeddings on your own dataset, and determine
       embedding quality through intrinsic evaluation.
 
+   .. card::
+      :title: Extracting Sentence Features with Pre-trained ELMo
+      :link: sentence_embedding/elmo_sentence_representation.html
+
+      See how to use GluonNLP's model API to automatically download the pre-trained ELMo
+      model from NAACL2018 best paper, and extract features with it.
```
Comment (eric-haibin-lin) [inline docs/examples/index.rst]: See how to use GluonNLP's model API to automatically download the pre-trained ELMo
model from NAACL2018 best paper, and extract features with it.
-> 
See how to use GluonNLP's API to automatically download the pre-trained ELMo
model, and extract features from it.

File: docs/examples/index.rst [hunk 2]
```diff
@@ -23,6 +23,20 @@ Word Embedding
...
+
+   .. card::
+      :title: Fine-tuning Pre-trained BERT Model
```
Comment (eric-haibin-lin) [inline docs/examples/index.rst]: Fine-tuning Pre-trained BERT Model ->
Fine-tuning Pre-trained BERT Models

File: docs/examples/index.rst [hunk 3]
```diff
@@ -127,13 +116,13 @@ Sentiment Analysis
...
-Sequence Sampling
------------------
+Text Generation
+---------------
 
 .. container:: cards
 
    .. card::
-      :title: Sequence Generation with Sampling and Beam Search
+      :title: Inference with Beam Search Sampler and Sequence Sampler
```
Comment (eric-haibin-lin) [inline docs/examples/index.rst]: Inference with Beam Search Sampler and Sequence Sampler
-> 
Sequence Generation with Beam Search Sampler and Sequence Sampler

File: docs/examples/sentiment_analysis/index.rst [hunk 4]
```diff
@@ -4,16 +4,24 @@ Sentiment Analysis
...
+
+
 
 .. toctree::
    :hidden:
    :maxdepth: 2
 
    sentiment_analysis.ipynb
-
+   ../sentence_embedding/self_attentive_sentence_embedding.ipynb
```
Comment (eric-haibin-lin) [inline docs/examples/sentiment_analysis/index.rst]: What about moving self_attentive_sentence_embedding.ipynb to the sentiment analysis folder?

**Comment by StrayBird-ATSH** [comment]
Yes. For the API notes, I plan to deal with them in the next step so that in these pull request, all the changes are concerning existing examples files, for more simplicity.
I will change the .md files accordingly. Thanks for your remind.

**Comment by mli** [comment]
Job PR-1154/10 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1154/10/index.html

**Comment by StrayBird-ATSH** [comment]
@eric-haibin-lin For your review. 
Thanks!

**Comment by eric-haibin-lin** [comment]
Good job. Look forward to the community page and data/vocab API notes

**Pull request merged**